extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboard.ts

```diff
@@ -0,0 +1,223 @@
+
+			// Service endpoints header row
+			const endpointsHeaderRow = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'row' }).component();
+			const endpointsServiceNameHeaderCell = view.modelBuilder.text().withProperties<azdata.TextComponentProperties>({ value: localize('bdc.dashboard.serviceHeader', "Service") }).component();
+			endpointsHeaderRow.addItem(endpointsServiceNameHeaderCell, { CSSStyles: { 'width': '25%', 'font-weight': 'bold', 'user-select': 'text' } });
+			const endpointsEndpointHeaderCell = view.modelBuilder.text().withProperties<azdata.TextComponentProperties>({ value: localize('bdc.dashboard.endpointHeader', "Endpoint") }).component();
+			endpointsHeaderRow.addItem(endpointsEndpointHeaderCell, { CSSStyles: { 'width': '15%', 'font-weight': 'bold', 'user-select': 'text' } });
+			endpointsContainer.addItem(endpointsHeaderRow, { CSSStyles: { 'padding-left': '10px', 'box-sizing': 'border-box', 'user-select': 'text' } });
+
+			// Service endpoints rows
+			this.model.serviceEndpoints.forEach(endpointInfo => {
+				const endPointRow = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'row', alignItems: 'center', height: '20px' }).component();
+				const nameCell = view.modelBuilder.text().withProperties<azdata.TextComponentProperties>({ value: getFriendlyEndpointNames(endpointInfo.serviceName) }).component();
+				endPointRow.addItem(nameCell, { CSSStyles: { 'width': '25%', 'user-select': 'text' } });
+				if (endpointInfo.isHyperlink) {
+					const linkCell = view.modelBuilder.hyperlink().withProperties<azdata.HyperlinkComponentProperties>({ label: endpointInfo.hyperlink, url: endpointInfo.hyperlink }).component();
+					endPointRow.addItem(linkCell, { CSSStyles: { 'width': '35%', 'color': '#0078d4', 'text-decoration': 'underline', 'overflow': 'hidden' } });
+				}
+				else {
+					const endpointCell = view.modelBuilder.text().withProperties<azdata.TextComponentProperties>({ value: endpointInfo.ipAddress + ':' + endpointInfo.port }).component();
+					endPointRow.addItem(endpointCell, { CSSStyles: { 'width': '35%', 'user-select': 'text', 'overflow': 'hidden' } });
+				}
+				const copyValueCell = view.modelBuilder.button().component();
+				copyValueCell.iconPath = this.copyIconPath;
+				copyValueCell.onDidClick(() => {
+					vscode.env.clipboard.writeText(endpointInfo.hyperlink);
+				});
+				copyValueCell.title = localize("bdc.dashboard.copyTitle", "Copy");
+				copyValueCell.iconHeight = '14px';
+				copyValueCell.iconWidth = '14px';
+				endPointRow.addItem(copyValueCell, { CSSStyles: { 'width': '5%', 'padding-left': '10px' } });
+
+				endpointsContainer.addItem(endPointRow, { CSSStyles: { 'padding-left': '10px', 'border-top': 'solid 1px #ccc', 'box-sizing': 'border-box', 'user-select': 'text' } });
+			});
+
+			contentContainer.addItem(endpointsContainer);
+
+			await view.initializeModel(dashboardRootContainer);
+		});
+	}
+}
+
+function getFriendlyEndpointNames(name: string): string {
```

Comment:
It's frustrating we have to duplicate this stuff from the mssql extension. Note I'm making changes that'll impact this (there are new endpoints we need to add, and logic on which should be hyperlinked). Can we check with Anthony and see if (in the future, not this PR) we can more easily share some component logic between extensions?

Reply:
Agreed. I think the best option for this is to have a core extension (mssql) export functionality like this. That's pretty easy to set up. I'll create an issue for tracking this next release.